sql/core/src/test/java/test/org/apache/spark/sql/JavaDataFrameSuite.java

```diff
@@ -258,7 +258,9 @@ public void testSampleBy() {
     DataFrame df = context.range(0, 100, 1, 2).select(col("id").mod(3).as("key"));
     DataFrame sampled = df.stat().<Integer>sampleBy("key", ImmutableMap.of(0, 0.1, 1, 0.2), 0L);
     Row[] actual = sampled.groupBy("key").count().orderBy("key").collect();
-    Row[] expected = {RowFactory.create(0, 5), RowFactory.create(1, 8)};
-    Assert.assertArrayEquals(expected, actual);
+    Assert.assertEquals(0, actual[0].getLong(0));
+    Assert.assertTrue(1 <= actual[0].getLong(1) && actual[0].getLong(1) <= 6);
+    Assert.assertEquals(1, actual[1].getLong(0));
+    Assert.assertTrue(4 <= actual[1].getLong(1) && actual[1].getLong(1) <= 11);
```

Comment:
If it's more complicated than just checking for the same range you do here, leave it. If it's really the same thing and just a matter of adding a range check, probably worth it.